pylint/checkers/variables.py

```diff
@@ -2210,34 +2210,31 @@ def _is_variable_violation(
                 and isinstance(defstmt.parent, nodes.If)
                 and in_type_checking_block(defstmt)
                 and not in_type_checking_block(node)
+                and not VariablesChecker._is_variable_annotation_in_function(node)
+                and not defstmt.parent.parent_of(node)
```

Comment:
Although it was here before, nothing fails when I comment out `and not defstmt.parent.parent_of(node)`. So that's a good candidate for a test case.